.github/workflows/test_build.yml

```diff
@@ -25,143 +25,188 @@ on:
+        run: |
+          cmake --build . --target build-libraries build-programs
+
+      - name: Build Macaulay2 using Ninja
+        if: matrix.build-system == 'cmake'
         run: |
-          cmake --build . --target build-libraries  -j4
-          cmake --build . --target build-programs   -j4
-          cmake --build . --target M2-core          -j4
-          cmake --build . --target install-packages -j4
+          cmake --build . --target M2-core install-packages
 
 # ----------------------
-#   Configure and build M2 the old way
+#   Configure and build M2 using Autotools
 # ----------------------
 
-      - name: Configure Macaulay2 (autotools)
-        id: step_configure_autotools
-        if: ${{ matrix.build-type == 'autotools' }}
-        env:
-          CXX: ${{matrix.cxx}}
-          CC: ${{matrix.cc}}
+      - name: Configure Macaulay2 using Autotools
+        if: matrix.build-system == 'autotools'
         run: |
-          cd M2
-          ${{matrix.make}} get-tools
-          ${{matrix.make}} -f Makefile
-          cd BUILD/cicd
-          ../../configure --prefix=/usr --enable-download --enable-build-libraries="readline"
-
-      - name: Save artifacts (autotools)
-        if: always() && matrix.build-type == 'autotools'
-        uses: actions/upload-artifact@v1
-        with:
-           name: config-log-${{matrix.os}}-${{matrix.compiler}}
-           path: M2/BUILD/cicd/config.log
+          make -C ../.. get-libtool get-automake get-autoconf
+          make -C ../.. all
+          ../../configure --enable-download
 
-      - name: Build Macaulay2 (autotools)
-        id: step_build_autotools
-        if: ${{ matrix.build-type == 'autotools' }}
-        working-directory: M2/BUILD/cicd
-        run: ${{matrix.make}}
+      - name: Build Macaulay2 using Make
+        if: matrix.build-system == 'autotools'
+        run: |
+          make
 
 # ----------------------
 #   Run tests
 # ----------------------
 
-      - name: Run Tests (autotools, ubuntu)
-        if: matrix.build-type == 'autotools' && matrix.os == 'ubuntu-latest'
-        working-directory: M2/BUILD/cicd
+      - name: Run Tests using CTest
+        if: matrix.build-system == 'cmake' && matrix.os == 'ubuntu-latest'
         run: |
-          ${{matrix.make}} check
-          ${{matrix.make}} -C Macaulay2/html-check-links check
-          ${{matrix.make}} validate-html
+          # cmake --build . --target M2-unit-tests
+          # ctest -R "unit-tests"
+          ctest -R "normal/" -E "command|program|schenck|threads|doc-links" -j4
+
+      - name: Run Tests using Autotools
+        if: matrix.build-system == 'autotools' && matrix.os == 'ubuntu-latest'
+        run: |
+          make check
+          make -C Macaulay2/html-check-links check
+          make validate-html
+
+# ----------------------
+#   Upload build logs
+# ----------------------
+
+      - name: Upload build logs
+        if: failure()
```

Comment:
We probably want the build logs always, not just after a failure.  For example, what if we want to determine why a configure script didn't detect a certain package?